pkg/aws/common_mutator.go

```diff
@@ -101,111 +97,7 @@ func GetReleaseComponentLabels(release releasev1alpha1.Release) map[string]strin
-	if clusterID == "" {
-		return nil, microerror.Maskf(invalidConfigError, "Object has no %s label, can't fetch AWSCluster.", label.Cluster)
-	}
-
-	// Fetch the AWSCluster CR
-	{
-		m.Logger.Log("level", "debug", "message", fmt.Sprintf("Fetching AWSCluster %s", clusterID))
-		fetch = func() error {
-			err := m.K8sClient.CtrlClient().Get(context.Background(), client.ObjectKey{Name: clusterID, Namespace: namespace}, &awsCluster)
-			if IsNotFound(err) {
-				return microerror.Maskf(notFoundError, "Looking for AWSCluster named %s but it was not found.", clusterID)
-			} else if err != nil {
-				return microerror.Mask(err)
-			}
-			return nil
-		}
-	}
-
-	{
-		b := backoff.NewMaxRetries(3, 10*time.Millisecond)
-		err = backoff.Retry(fetch, b)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-	}
-	return &awsCluster, nil
-}
-
-func FetchCluster(m *Mutator, meta metav1.Object) (*capiv1alpha2.Cluster, error) {
-	var cluster capiv1alpha2.Cluster
-	var err error
-	var fetch func() error
-
-	namespace := meta.GetNamespace()
-	if namespace == "" {
-		namespace = metav1.NamespaceDefault
-	}
-	// Retrieve the Cluster ID.
-	clusterID := key.Cluster(meta)
-	if clusterID == "" {
-		return nil, microerror.Maskf(invalidConfigError, "Object has no %s label, can't fetch cluster.", label.Cluster)
-	}
-
-	// Fetch the Cluster CR
-	{
-		m.Logger.Log("level", "debug", "message", fmt.Sprintf("Fetching Cluster %s", clusterID))
-		fetch = func() error {
-			err := m.K8sClient.CtrlClient().Get(context.Background(), client.ObjectKey{Name: clusterID, Namespace: namespace}, &cluster)
-			if IsNotFound(err) {
-				return microerror.Maskf(notFoundError, "Looking for Cluster named %s but it was not found.", clusterID)
-			} else if err != nil {
-				return microerror.Mask(err)
-			}
-			return nil
-		}
-	}
-
-	{
-		b := backoff.NewMaxRetries(3, 10*time.Millisecond)
-		err = backoff.Retry(fetch, b)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-	}
-	return &cluster, nil
-}
-
-func FetchRelease(m *Mutator, version *semver.Version) (*releasev1alpha1.Release, error) {
-	var releaseName string
-	var release releasev1alpha1.Release
-	var err error
-
-	// Get release name
-	{
-		releaseName = version.String()
-		if !strings.HasPrefix(releaseName, "v") {
-			releaseName = fmt.Sprintf("v%s", releaseName)
-		}
-	}
-	// Fetch the Release CR
-	{
-		m.Logger.Log("level", "debug", "message", fmt.Sprintf("Fetching Release %s", releaseName))
-		err = m.K8sClient.CtrlClient().Get(context.Background(), client.ObjectKey{Name: releaseName, Namespace: metav1.NamespaceDefault}, &release)
-		if IsNotFound(err) {
-			return nil, microerror.Maskf(notFoundError, "Looking for Release %s but it was not found.", releaseName)
-		} else if err != nil {
-			return nil, microerror.Mask(err)
-		}
-	}
-	return &release, nil
-}
-
-func GetNavailabilityZones(m *Mutator, n int, azs []string) []string {
+func GetNavailabilityZones(m *Handler, n int, azs []string) []string {
```

Comment:
Maybe name it to 

```suggestion
func FetchNavailabilityZones(m *Handler, n int, azs []string) []string {
```

and then move it to common_fetcher.go to be a bit more consistent?

Reply:
Hmmm I think this function doesn't really fit with the others in `common-fetcher` since it doesn't fetch another CR but rather just compute something that is used for the mutator.